pkg/infra/kubernetes/exec_unit.go

```diff
@@ -76,197 +125,165 @@ func shouldTransformServiceAccount(unit *core.ExecutionUnit) bool {
 	return shouldTransformImage(unit)
 }
 
-func (unit *HelmExecUnit) transformPod(cfg config.ExecutionUnit) (values []HelmChartValue, err error) {
-	log := zap.L().Sugar().With(logging.FileField(unit.Pod), zap.String("unit", unit.Name))
-	log.Debugf("Transforming file, %s, for exec unit, %s", unit.Pod.Path(), unit.Name)
-	obj, err := readFile(unit.Pod)
-	if err != nil {
-		return
-	}
-	pod, ok := obj.(*corev1.Pod)
-	if !ok {
-		err = fmt.Errorf("expected file %s to contain Pod Kind", unit.Pod.Path())
-		return
-	}
+var podTransformer = manifestTransformer[*corev1.Pod]{
+	fieldToTransform: func(unit *HelmExecUnit) *core.SourceFile {
+		return unit.Pod
+	},
 
-	_, imagePlaceholder, err := unit.upsertOnlyContainer(&pod.Spec.Containers, cfg)
-	if err != nil {
-		return nil, err
-	}
+	transform: func(unit *HelmExecUnit, cfg config.ExecutionUnit, pod *corev1.Pod, log *zap.SugaredLogger) ([]HelmChartValue, error) {
+		_, imagePlaceholder, err := unit.upsertOnlyContainer(&pod.Spec.Containers, cfg)
+		if err != nil {
+			return nil, err
+		}
 
-	if pod.Labels == nil {
-		pod.Labels = make(map[string]string)
-	}
-	pod.Labels["execUnit"] = unit.Name
-	pod.Spec.ServiceAccountName = unit.getServiceAccountName()
+		if pod.Labels == nil {
+			pod.Labels = make(map[string]string)
+		}
+		pod.Labels["execUnit"] = unit.Name
+		pod.Spec.ServiceAccountName = unit.getServiceAccountName()
 
-	output, err := yaml.Marshal(pod)
-	if err != nil {
-		return
-	}
-	err = unit.Pod.Reparse([]byte(output))
-	if err != nil {
-		return
-	}
-	values = append(values, HelmChartValue{
-		ExecUnitName: unit.Name,
-		Kind:         pod.Kind,
-		Type:         string(ImageTransformation),
-		Key:          imagePlaceholder,
-	})
-	return
+		output, err := yaml.Marshal(pod)
+		if err != nil {
+			return nil, err
+		}
+		err = unit.Pod.Reparse([]byte(output))
+		if err != nil {
+			return nil, err
+		}
+		return []HelmChartValue{
+			HelmChartValue{
```

Comment:
I think linter would be unhappy about this (the `HelmChartValue` can be omitted)

Reply:
We do run `golangci-lint`, and it didn't complain, but I can rm it.

Reply:
Huh, nevermind then. I swear I've done similar things and it complained to me. 🤷